File: ClientApp/src/app/model/RoomDto.ts
import { Deck } from "./Deck";
import { RoomState } from "./Room";
import { Card, Game, Sentence } from "./Game";
import { User } from "./User";

export interface RoomDto {
  id: string;
  name: string;
  ownerId: string;
  ownerName: string;
  maxPlayers: number;
  users: User[];
  bannedUsers: string[];
  decks: Deck[];
  sentences: Sentence[];
  cards: Card[];
  state: RoomState;
  game?: Game | null;
}

File: Model/DTO/RoomDTO.cs
using CardsAngular.Model.Entities;
using CardsAngular.Model.Enumarations;

namespace CardsAngular.Model.DTO;

public record RoomDTO(
    Guid Id,
    string? Name,
    Guid OwnerId,
    string OwnerName,
    int MaxPlayers,
    IEnumerable<UserDTO> Users,
    IEnumerable<Guid> BannedUsers,
    IEnumerable<DeckDTO> Decks,
    IEnumerable<Sentence> Sentences,
    IEnumerable<Card> Cards,
    RoomState State,
    GameDTO? Game
);